packages/lodestar/src/db/api/beacon/repository.ts

```diff
@@ -64,12 +64,18 @@ export abstract class BulkRepository<T> extends Repository<T> {
     return (data || []).map((data) => deserialize(data, this.type));
   }
 
-  public async getAllBetween(lowerLimit: number|null, upperLimit: number|null): Promise<T[]> {
+  public async getAllBetween(lowerLimit: number | null, upperLimit: number | null, step?: number | null): Promise<T[]> {
     const data = await this.db.search({
       gt: encodeKey(this.bucket, lowerLimit || Buffer.alloc(0)),
       lt: encodeKey(this.bucket, upperLimit || Number.MAX_SAFE_INTEGER),
     });
-    return (data || []).map((data) => deserialize(data, this.type));
+    let processedData = data;
+    if (typeof step !== "undefined" && step !== null) {
+      processedData = data.filter((datum, index) => {
+        return index % step === 0;
+      });
+    }
```

Comment:
This is kinda awkward here and we are reading a lot of data that we dont need. Also this allows us to do same thing in other Buckets.

I would suggest to add this code here: https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar/src/db/controller/impl/level.ts#L89

You can do this if you add optional param to ISearchOptions.